src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -0,0 +1,163 @@
+import _isEmpty from 'lodash/isEmpty';
+import _isNil from 'lodash/isNil';
+import _isObject from 'lodash/isObject';
+import PositionModel from '../base/PositionModel';
+import RouteModel from '../airport/Route/RouteModel';
+import {
+    fixRadialDist,
+    isWithinAirspace,
+    calculateDistanceToBoundary,
+    bearingToPoint
+} from '../math/flightMath';
+import { nm } from '../utilities/unitConverters';
+
+/**
+ * Loop through `waypointModelList` and determine where along the route an
+ * aircraft should spawn
+ *
+ * @function _calculateSpawnPositions
+ * @param waypointModelList {array<StandardWaypointModel>}
+ * @param spawnOffsets {array}
+ * @return spawnPositions {array}
+ */
+export const _calculateSpawnPositions = (waypointModelList, spawnOffsets, airport) => {
```

Comment:
another good question!

It's actually simpler than that.  I'm exporting these so I can test them.  But, because they are all called by a public method, testing them is slightly overkill.  In fact, I don't think I actually _did_ test them because I was able to get to 100% without test each function.

If thats the case, I will likely remove the export because it is confusing.